src/main/java/com/mongodb/jdbc/MongoResultSetMetaData.java

```diff
@@ -503,10 +521,122 @@ public boolean isDefinitelyWritable(int column) throws SQLException {
     // --------------------------JDBC 2.0-----------------------------------
 
     @Override
+	// TODO
     public String getColumnClassName(int column) throws SQLException {
         checkBounds(column);
-        Object o = mongoResultDoc.values.get(column - 1).value;
-        return o.getClass().getName();
```

Comment:
It turns out that the previous implementation of this function was incorrect. It's supposed to return the name of the class that would be returned by `resultSet.getObject()`, which does _not_ return BSON values

Reply:
Great catch